vendor/k8s.io/kube-openapi/pkg/handler3/handler.go

```diff
@@ -96,10 +122,17 @@ func (o *OpenAPIService) getGroupBytes() ([]byte, error) {
 	}
 
 	sort.Strings(keys)
-	group := make(map[string][]string)
-	group["Paths"] = keys
-
-	j, err := json.Marshal(group)
+	discovery := &OpenAPIV3Discovery{Paths: make(map[string]OpenAPIV3DiscoveryGroupVersion)}
+	for gvString, groupVersion := range o.v3Schema {
+		etagBytes, err := groupVersion.etagCache.Get()
```

Comment:
This is assuming the kube-aggregator method of computing the etag will always match aggregated servers' method of computing etag (because we're embedding etags into url params in the root discovery doc, but then requiring them to match in the actual openapi handler). That seems fragile, and means we can't change how etags are computed in the future.

Reply:
For OpenAPI V3, the aggregator will proxy requests to aggregated servers' rather than downloading and recomputing etags/etc.

Reply:
ah, ok, I missed that

Reply:
oh, and it will just aggregate their discovery docs, including whatever hashes the backends include?